crates/holochain/src/conductor/cell/validation_package.rs

```diff
@@ -36,14 +43,126 @@ impl ValidationPackageDb {
     }
 }
 
+#[instrument(skip(header_hashed, env, ribosome, conductor_api, network))]
 pub(super) async fn get_as_author(
-    header: Header,
+    header_hashed: HeaderHashed,
     env: EnvironmentRead,
-    dna_file: &DnaFile,
+    ribosome: &impl RibosomeT,
     conductor_api: &impl CellConductorApiT,
+    network: &HolochainP2pCell,
 ) -> CellResult<ValidationPackageResponse> {
+    let header = header_hashed.as_content();
+
     // Get the source chain with public data only
-    let source_chain = SourceChain::public_only(env)?;
+    let source_chain = SourceChain::public_only(env.clone())?;
+
+    // Get the header data
+    let (app_entry_type, header_seq) = match header
+        .entry_type()
+        .cloned()
+        .map(|et| (et, header.header_seq()))
+    {
+        Some((EntryType::App(aet), header_seq)) => (aet, header_seq),
+        _ => return Ok(None.into()),
+    };
+
+    //Get entry def
+    let entry_def = get_entry_def_from_ids(
+        app_entry_type.zome_id(),
+        app_entry_type.id(),
+        ribosome.dna_file(),
+        conductor_api,
+    )
+    .await?;
+
+    // Get the required validation package
+    let required_validation_type = match entry_def {
+        Some(ed) => ed.required_validation_type,
+        None => return Ok(None.into()),
+    };
+
+    // Gather the package
+    match required_validation_type {
+        RequiredValidationType::Element => {
+            // TODO: I'm not sure if we should handle this case, it seems like they should already have the element
+            Ok(None.into())
+        }
+        RequiredValidationType::SubChain => Ok(Some(get_as_author_sub_chain(
+            header_seq,
+            app_entry_type,
+            &source_chain,
+        )?)
+        .into()),
+        RequiredValidationType::Full => {
+            Ok(Some(get_as_author_full(header_seq, &source_chain)?).into())
+        }
+        RequiredValidationType::Custom => {
```

Comment:
I'm not exactly sure what you mean by "this" but I think you mean the whole block for this arm.
The reason it can't be inside `get_as_author_custom` is because there's differences between how you construct the cascade here vs where it's used directly in direct app validation because there it needs to use the existing workspace so that the scratch space is available.
Imagine the scenario where you commit two entries in one call and each entry requires the full chain. In that case some of the validation package is only available in the scratch.

Reply:
@freesig oops I meant to delete this comment after I read further in the PR but never did :see_no_evil: